nix/module.nix

```diff
@@ -1,18 +1,14 @@
-self: {
-  config,
-  lib,
-  pkgs,
-  ...
-}:
-with lib; let
+{ config, lib, pkgs, inputs ? builtins.throw "Expected an inputs argument that corresponds to flake conventions", ... }: #TODO not sure this default arg is a good solution to this
+let
   cfg = config.services.liberaforms;
+  packages = inputs.self.packages.${pkgs.system};
```

Comment:
This will almost always result in double nixkpgs evaluation. Please use overlays instead.

I'd prefer seeing a separate overlay where users themselves need to wire it instead of using `nixpkgs.overlays` but its your choice

Reply:
Why would it result in a double eval?

Reply:
modules system reimports nipxkgs undless `nixpkgs.pkgs` option is set

Reply:
Nice catch.

I didn't notice this, but re: I didn't really look at the module.

In this case you are correct about there being two evaluations, but it's still not the module system's fault.

In this case the double import happens because here
- the flake package set that depends on the nixpkgs defined in flake inputs is used to build the packages, 
- whereas when the module is imported, it's already using a nixpkgs that comes from original evaluation of the module system, i.e. things originating from two different scopes with two different nixpkgs evaluations are used.

So indeed using an overlay would make sure the package are evaluated with the nixpkgs being used in the module system.